### PR TITLE
Fix for #76

### PR DESCRIPTION
**Pull request by omangin**
There is actually a little more than the fix (see commit messages for more details).

Note: I didn't add any non-regression test at the usecase level because these things are not really caught by the `fakefilesystem` thing. It might worth spending a little time checking for these or eventually adding a few assertions directly in the `write_file` method from `content.py` to make sure that it's called properly (i.e. no binary content without binary mode). The latter is probably the quickest way to check that.

### CODE REVIEW

File: pubs/content.py [hunk 1]
```diff
@@ -52,16 +52,17 @@ def _open(path, mode):
     else:
```
Comment (benureau) [inline pubs/content.py]: I like 'Cosmit' a lot.

Comment (omangin) [inline pubs/content.py]: I should have called it PEP8 ;)

Is this comment related in any way to the line it's attached to?

Comment (benureau) [inline pubs/content.py]: No no no. To the title of you commit. But I did not find any way to comment that directly.

**Comment by benureau** [comment]
@omangin: I was wondering about adding things like DOI tests in the usecases, but that would mean the test would depend on access to the internet. What do you think?

**Comment by omangin** [comment]
> @omangin: I was wondering about adding things like DOI tests in the usecases, but that would mean the test would depend on access to the internet. What do you think?
 
I was worried about the tests needing a connection to work but I think it's worth it to have the tests, so that we'd also be noticed if the APIs change. I'll add a few tests before merging.
